### PR TITLE
pydantic utils

### PR DESCRIPTION
**Pull request by aaazzam**

### CODE REVIEW

**Comment by jlowin** [comment]
How does this utility compare to the one here: https://github.com/PrefectHQ/marvin/blob/main/src/marvin/utilities/types.py#L339-L370

File: src/marvin/utilities/pydantic.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+def create_model_from_schema(schema: dict):
+    # create a temporary json file
+    name_of_model = schema.get('title', 'Model')
+
+    try:
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as f:
+            json.dump(schema, f)
+            temp_json_file = f.name
+            f.close()
```
Comment (zzstoatzz) [inline src/marvin/utilities/pydantic.py:23]: why are we calling `f.close()`?

File: src/marvin/utilities/pydantic.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        with tempfile.NamedTemporaryFile(mode='w+t', delete=False) as f:
+            generate(
+                input_=Path(temp_json_file),
+                input_file_type=InputFileType.JsonSchema,
+                output=Path(f.name + '.py'),
+            )
+            temp_python_file = f.name + '.py'
+            f.close()
```
Comment (zzstoatzz) [inline src/marvin/utilities/pydantic.py:32]: same as above, and the rest below

File: src/marvin/utilities/pydantic.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            file.close()
+ 
+        spec = importlib.util.spec_from_file_location("temp_module", temp_python_file)
+
+        # Create a module from the spec
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+        import inspect
+
+    finally:
```
Comment (zzstoatzz) [inline src/marvin/utilities/pydantic.py:49]: likewise, wouldnt the tmp context manager clean this up already?

**Pull request closed without merging**